### PR TITLE
Improvement: Improve project file structure

### PR DESCRIPTION
**Pull request by jane-rose**
* Fix: Fix bug on toolpath params display on cnc part
* Improvement: Add toolpath on project file and change project file structure

### CODE REVIEW

File: resources/CuraEngine/Config/CncConfig/Default.def.json [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {
     "category": "Default",
     "definitionId": "Default",
+    "version": '1.0.0',
```
Comment (parachvte) [inline resources/CuraEngine/Config/CncConfig/Default.def.json:4]: use "

File: src/app/flux/project/index.js [hunk 2]
```diff
@@ -231,9 +259,10 @@ export const actions = {
             }
             formData.append('file', file);
             const { body: { content } } = await api.recoverProjectFile(formData);
-            const envObj = JSON.parse(content);
-            const headType = envObj.headType;
 
+            const envObj = JSON.parse(content);
```
Comment (parachvte) [inline src/app/flux/project/index.js:263]: Add validation for env object? Never trust the input.

File: src/app/views/ToolPathConfigurations/ToolPathConfigurations.jsx [hunk 3]
```diff
@@ -104,16 +135,29 @@ class ToolPathConfigurations extends PureComponent {
...
+        return !Object.entries(currentToolDefinition.config).every(([key, setting]) => {
+            return activeToolDefinition.config[key].default_value === setting.default_value;
+        });
+    }
+
     render() {
         if (!this.state.toolPath) {
             return null;
         }
+        const isModifiedDefinition = this.checkIfDefinitionModified();
```
Comment (parachvte) [inline src/app/views/ToolPathConfigurations/ToolPathConfigurations.jsx:160]: Potential performance issue here. If you insist on this, add a comment to indicate the issue.

File: src/app/flux/project/index.js [hunk 4]
```diff
@@ -136,6 +147,23 @@ export const actions = {
...
+                    ...toolpaths[k],
+                    modelGroup
+                });
+                toolPathGroup.toolPaths.push(nToolPath);
+                toolPathGroup.toolPathObjects.add(nToolPath.object);
+
+                toolPathGroup.selectedToolPathId = nToolPath.id;
+                nToolPath.commitGenerateToolPath({ materials });
+            }
+        }
```
Comment (zjiefee) [inline src/app/flux/project/index.js:166]: Generate Model 的时候会生成新的 ModelID，与 toolpath 里面保存的对应不上

File: src/app/views/ToolPathConfigurations/ToolPathConfigurations.jsx [hunk 5]
```diff
@@ -88,6 +88,37 @@ class ToolPathConfigurations extends PureComponent {
...
+                            config.plunge_speed.default_value = gcodeConfig.plungeSpeed;
+                        }
+                        if (config.work_speed.default_value !== gcodeConfig.workSpeed) {
+                            config.work_speed.default_value = gcodeConfig.workSpeed;
+                        }
+                        if (config.step_down.default_value !== gcodeConfig.stepDown) {
+                            config.step_down.default_value = gcodeConfig.stepDown;
+                        }
+                        // if (config.density.default_value !== gcodeConfig.density) {
+                        // }
```
Comment (zjiefee) [inline src/app/views/ToolPathConfigurations/ToolPathConfigurations.jsx:115]: 密度需要出来

**Comment by parachvte** [comment]
Reference internal JIRA LU-633

**Pull request merged**